### PR TITLE
Create Client API v2.0 for JavaScript SDK

### PR DESCRIPTION
**Pull request by makesitgo**

### CODE REVIEW

File: src/services/mongodb/collection.js [hunk 1]
```diff
@@ -143,109 +143,56 @@ Collection.prototype.upsert =
 
 // private
 function insertOp(self, docs, options) {
-  let stages = [];
-
-  // there may be no docs, when building for chained pipeline stages in
-  // which case the source is considered to be the previous stage
-  if (docs) {
```
Comment (makesitgo) [inline src/services/mongodb/collection.js]: [ ⚠️  ]: no longer leaving the potential for no `docs` to be provided to this method, as chaining stages is no longer supported

File: test/admin/dev.test.js [hunk 2]
```diff
@@ -39,37 +47,37 @@ describe('Dev Function', () => {
...
-      user._id,
+    const result = await client.executeFunction(
       FUNC_NAME,
       777,
       23,
     );
 
-    expect(result.result.sum).toEqual({'$numberDouble': '800'});
-    expect(result.result.userId).toEqual(user._id);
+    expect(result.sum).toEqual(800);
```
Comment (makesitgo) [inline test/admin/dev.test.js]: [ ⚠️  ]: calling out this test assertion change

Comment (edaniels) [reply test/admin/dev.test.js]: 800 because of decoding extended json?

**Comment by edaniels** [comment]
@makesitgo you should rebase off of master and set the PR to be based on that

File: test/admin/dev.test.js (hunk 2 above)
Comment (makesitgo) [reply test/admin/dev.test.js]: @edaniels ... I'm not entirely sure why I needed to change this, but simply made it based off what I was seeing come in/out from the Stitch backend.

Comment (edaniels) [reply test/admin/dev.test.js]: Cool

Comment (makesitgo) [reply test/admin/dev.test.js]: just to tie off loose ends, this change was _needed_ (read: not actually needed) b/c I inadvertently switched the test from using the admin api to the client api.  code has been reverted to use the proper api in #84

**Pull request closed without merging**